registry/extension/oras/artifactservice.go

```diff
@@ -72,7 +89,16 @@ func (h *referrersHandler) Referrers(ctx context.Context, revision digest.Digest
 		return nil, err
 	}
 
-	return referrers, nil
+	// sort the list of descriptors that contain the created annotation
+	sort.Slice(referrersSorted, func(i, j int) bool {
+		firstElem, _ := time.Parse(time.RFC3339, referrersSorted[i].Annotations[createAnnotationName])
+		secondElem, _ := time.Parse(time.RFC3339, referrersSorted[j].Annotations[createAnnotationName])
+		// most recent artifact first
+		return firstElem.After(secondElem)
+	})
+	// append the descriptors, which don't have a created annotation, to the end
+	referrersSorted = append(referrersSorted, referrersUnsorted...)
+	return referrersSorted, nil
```

Comment:
This now may also contain a whole lot of annotations. Do we want to limit the maximum size of the returned response?

Reply:
I have removed the annotations from the referrers response since the [referrers spec](https://github.com/oras-project/artifacts-spec/blob/main/manifest-referrers-api.md#artifact-referrers-api-results) sample response does not contain the field. There's also a concern that a large number of annotations can cause response size issues.